src/services/tech-record-document-conversion.ts

```diff
@@ -241,6 +243,24 @@ const upsertVehicle = async (
   connection: Connection,
   techRecordDocument: TechRecordDocument,
 ): Promise<number> => {
+
+  if (techRecordDocument.techRecord && techRecordDocument.techRecord[0].statusCode === 'archived') {
+      debugLog('upsertTechRecords: Retrieving vehicle...');
+      // retrieve records associated with systemNumber
+      const existingRecordIds = await selectRecordIds(
+          VEHICLE_TABLE.tableName,
+          { system_number: techRecordDocument.systemNumber, vin: vinCleanser(techRecordDocument.vin) },
+          connection,
+      );
+      if (existingRecordIds.rows.length > 0) {
```

Comment:
Since you've mentioned that there's only ever going to be 0 or 1 row, I think it's worth being more specific here.
if rows length = 1 return...... else log..... 
just incase for whatever reason, it returns more than 1 row.